chia/_tests/core/consensus/test_mmr.py

```diff
@@ -0,0 +1,204 @@
+from __future__ import annotations
+
+import logging
+import os
+import random
+import time
+
+import pytest
+from chia_rs import FullBlock
+from chia_rs.sized_bytes import bytes32
+
+from chia.consensus.mmr import MerkleMountainRange, get_height, get_peak_positions, verify_mmr_inclusion
+
+logger = logging.getLogger(__name__)
+
+
+def random_bytes32() -> bytes32:
+    return bytes32(os.urandom(32))
+
+
+def test_empty_mmr_root() -> None:
+    mmr = MerkleMountainRange()
+    assert mmr.get_root() is None  # Empty MMR returns None
+    # Note: get_inclusion_proof not yet implemented
+    # assert mmr.get_inclusion_proof(random_bytes32()) is None
+
+
+def test_single_leaf() -> None:
+    mmr = MerkleMountainRange()
+    leaf = random_bytes32()
+    mmr.append(leaf)
+    root = mmr.get_root()
+    assert root is not None
+    proof = mmr.get_inclusion_proof_by_index(0)
+    assert proof is not None
+    peak_index, proof_bytes, other_peaks, peak = proof
+    assert verify_mmr_inclusion(root, leaf, peak_index, proof_bytes, other_peaks, peak)
+
+
+def test_multiple_leaves() -> None:
+    mmr = MerkleMountainRange()
+    leaves = [random_bytes32() for _ in range(10)]
+    for leaf in leaves:
+        mmr.append(leaf)
+    root = mmr.get_root()
+    assert root is not None
+    for idx, leaf in enumerate(leaves):
+        proof = mmr.get_inclusion_proof_by_index(idx)
+        assert proof is not None
+        peak_index, proof_bytes, other_peaks, peak = proof
+        assert verify_mmr_inclusion(root, leaf, peak_index, proof_bytes, other_peaks, peak)
+    # Root should be stable
+    mmr2 = MerkleMountainRange()
+    for leaf in leaves:
+        mmr2.append(leaf)
+    root2 = mmr2.get_root()
+    for i in range(10):
+        assert mmr.nodes[i] == mmr2.nodes[i]
+    assert root2 == root
```

Comment:
in addition to these round-trip tests for proofs, I think you should have a few concrete test cases that check that the resulting proof has exactly the nodes we expect it to have. Just to make sure the result is valid, not just that it validates in the validation function